src/Codegen/Generator.hack

```diff
@@ -495,22 +241,32 @@ final class Generator {
             $rc = new \ReflectionClass($enum->getName());
             $enum_type = $rc->getAttributeClass(\Slack\GraphQL\EnumType::class);
             if ($enum_type is null) continue;
-            $objects[] = new EnumInputType($enum, $enum_type);
-            $objects[] = new EnumOutputType($enum, $enum_type);
+            $objects[] = new InputEnumBuilder($enum_type, $enum->getName());
+            $objects[] = new OutputEnumBuilder($enum_type, $enum->getName());
         }
 
         // TODO: throw an error if no query fields?
 
         if (!C\is_empty($query_fields)) {
-            $top_level_objects = vec[new Query($query_fields)];
+            $top_level_objects = vec[
+                new ObjectBuilder(
+                    new \Slack\GraphQL\__Private\CompositeType('Query', 'Query'),
+                    'Slack\\GraphQL\\Root',
+                    $query_fields
+                )
+            ];
 
             if (!C\is_empty($mutation_fields)) {
-                $top_level_objects[] = new Mutation($mutation_fields);
+                $top_level_objects[] = new ObjectBuilder(
+                    new \Slack\GraphQL\__Private\CompositeType('Mutation', 'Mutation'),
+                    'Slack\\GraphQL\\Root',
+                    $mutation_fields
+                );
             }
 
             $objects = Vec\concat($top_level_objects, $objects, $inputs);
         }
 
-        return $objects;
+        return Vec\sort_by($objects, $object ==> $object->getGraphQLType());
```

Comment:
I think it's useful to keep `Query` and `Mutation` at the top of the file. I don't expect people to read the codegen very much but its a helpful guide to how the rest of it ties together.

Reply:
Will do. I like @jjergus ’s suggestion of breaking generated classes into separate files; will do that in a follow-up.